test/coreneuron/test_spikes.py

```diff
@@ -43,7 +43,10 @@ def test_spikes(use_mpi4py=False, use_nrnmpi_init=False, file_mode=False):
     # NEURON run
     nrn_spike_t = h.Vector()
     nrn_spike_gids = h.Vector()
-    pc.spike_record(-1, nrn_spike_t, nrn_spike_gids)
+
+    # rank 0 record spikes for all gid while others
+    # for specific gid. this is for better test coverage.
+    pc.spike_record(-1 if pc.id() == 0 else (pc.id()+1), nrn_spike_t, nrn_spike_gids)
```

Comment:
What makes you think there is an existing gid=pc.id()+1 on rank pc.id()?  For round robin distributions the gids are
pc.id() + i*pc.nhost()

Reply:
Good questions and I wanted to ask about this but I skimmed through the code in the test : https://github.com/neuronsimulator/nrn/blob/6dab82015dfeaa798119a2936e1945cd494a4b6d/test/coreneuron/test_spikes.py#L41

and thought we assign `pc.id()+1` as a gid using [cell](https://neuronsimulator.github.io/nrn/py_doc/modelspec/programmatic/network/parcon.html#ParallelContext.cell) method.

Reply:
Ok. I didn't realize this model was one cell per rank.